streamz/core.py

```diff
@@ -1231,6 +1231,59 @@ def cb(self):
             yield self._emit(x)
 
 
+@Stream.register_api()
+class to_kafka(Stream):
+    """ Writes data in the stream to Kafka
+
+    The Confluent client sends messages in batches and asynchronously. This
+    stream accepts a string or bytes object. Call ``flush`` to ensure all
+    messages are pushed. When a message is received, or in the event of an
+    error, this stream will emit a tuple in the form of (err, msg) see
+    https://github.com/confluentinc/confluent-kafka-python
+
+    Parameters
+    ----------
+    topic : string
+        The topic which to write
+    producer_config : dict
+        Settings to set up the stream, see
+        https://docs.confluent.io/current/clients/confluent-kafka-python/#configuration
+        https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+        Examples:
+        bootstrap.servers: Connection string(s) (host:port) by which to reach Kafka
+
+    Examples
+    --------
+    >>> source = Stream()
+    >>> source.map(lambda x: x + 1).to_kafka('test', {'bootstrap.servers': 'localhost'}).sink(
+    ...     lambda x: print(x[1].topic(), x[1].partition()))
+    >>> for i in range(3):
+    ...     source.emit(i)
+    test 0
+    test 0
+    test 0
+    """
+    Result = namedtuple('Result', ['err', 'msg'])
+
+    def __init__(self, upstream, topic, producer_config, **kwargs):
+        import confluent_kafka as ck
+
+        self.topic = topic
+        self.producer = ck.Producer(producer_config)
+
+        Stream.__init__(self, upstream, ensure_io_loop=True, **kwargs)
+
+    def update(self, x, who=None):
+        self.producer.poll(0)
```

Comment:
Is this necessary? If the data is not sent strictly synchronously, I think that's OK.

Reply:
I'm also curious about this, and more generally how well this approach works in an asynchronous environment.  It has been a while since I looked at confluent_kafka_python so I've forgotten some things

If the producer is busy then what happens with `producer.poll(0)`?  Does it err, does it block?  It may be that neither of these outcomes is desirable.